### PR TITLE
feat: add java solution to lc problem: No.1342 and No.1678

### PR DESCRIPTION
**Pull request by 0xcaffebabe**
feat: add java solution to lc problem: No.1342 and No.1678

### CODE REVIEW

**Comment by yanglbme** [comment]
Hi, @0xcaffebabe please create a Solution.java file

**Comment by yanglbme** [comment]
Refer to https://github.com/doocs/leetcode/pull/536/files

**Comment by 0xcaffebabe** [comment]
@yanglbme got it

**Comment by yanglbme** [comment]
please format your code before a commit

File: solution/1300-1399/1342.Number of Steps to Reduce a Number to Zero/README.md [hunk 1]
```diff
@@ -68,7 +68,17 @@
 <!-- 这里可写当前语言的特殊实现逻辑 -->
 
 ```java
-
+class Solution {
+    public int numberOfSteps(int num) {
+        int cnt = 0;
+        while(num != 0) {
```
Comment (yanglbme) [inline solution/1300-1399/1342.Number of Steps to Reduce a Number to Zero/README.md:74]:
```suggestion
        while (num != 0) {
```

File: solution/1300-1399/1342.Number of Steps to Reduce a Number to Zero/README_EN.md [hunk 2]
```diff
@@ -61,7 +61,17 @@ Step 4) 1 is odd; subtract 1 and obtain 0.
 ### **Java**
 
 ```java
-
+class Solution {
+    public int numberOfSteps(int num) {
+        int cnt = 0;
+        while(num != 0) {
```
Comment (yanglbme) [inline solution/1300-1399/1342.Number of Steps to Reduce a Number to Zero/README_EN.md:67]:
```suggestion
        while (num != 0) {
```

File: solution/1300-1399/1342.Number of Steps to Reduce a Number to Zero/Solution.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Solution {
+    public int numberOfSteps(int num) {
+        int cnt = 0;
+        while(num != 0) {
```
Comment (yanglbme) [inline solution/1300-1399/1342.Number of Steps to Reduce a Number to Zero/Solution.java:4]:
```suggestion
        while (num != 0) {
```

File: solution/1600-1699/1678.Goal Parser Interpretation/README.md [hunk 4]
```diff
@@ -64,7 +64,29 @@ G -&gt; G
 <!-- 这里可写当前语言的特殊实现逻辑 -->
 
 ```java
-
+class Solution {
+    public String interpret(String command) {
+        StringBuilder sb = new StringBuilder();
+        int p = 0,q = 1;
+        for(;p<command.length();p++,q++){
```
Comment (yanglbme) [inline solution/1600-1699/1678.Goal Parser Interpretation/README.md:71]:
```suggestion
        int p = 0, q = 1;
        for (; p < command.length(); p++, q++) {
```

**Review by yanglbme — CHANGES_REQUESTED** [review]

File: solution/1600-1699/1678.Goal Parser Interpretation/README_EN.md [hunk 5]
```diff
@@ -57,7 +57,29 @@ The final concatenated result is &quot;Goal&quot;.
 ### **Java**
 
 ```java
-
+class Solution {
+    public String interpret(String command) {
+        StringBuilder sb = new StringBuilder();
+        int p = 0,q = 1;
+        for(;p<command.length();p++,q++){
```
Comment (yanglbme) [inline solution/1600-1699/1678.Goal Parser Interpretation/README_EN.md:64]:
```suggestion
        int p = 0, q = 1;
        for (; p < command.length(); p++, q++) {
```

File: solution/1600-1699/1678.Goal Parser Interpretation/Solution.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class Solution {
+    public String interpret(String command) {
+        StringBuilder sb = new StringBuilder();
+        int p = 0,q = 1;
+        for(;p<command.length();p++,q++){
```
Comment (yanglbme) [inline solution/1600-1699/1678.Goal Parser Interpretation/Solution.java:5]:
```suggestion
        int p = 0, q = 1;
        for (; p < command.length(); p++, q++) {
```

**Review by yanglbme — CHANGES_REQUESTED** [review]

**Review by yanglbme — APPROVED** [review]

**Pull request merged**